### ISSUE TITLE
[CI] flaky test: not ok 19 runc run (cgroup v2 resources.unified override)

### ISSUE THREAD

**Issue by lifubang** [opencontainers/runc#4019]
### Description

After #3931 merged, we don't know why the container need more memory to run, so we open an new PR #3987 to reduce the memory usage when the container staring, but I think we may need a couple of days to review and merge it.
But now, there are 2 flaky tests in CI, because the limit of `memory.max`, it causes the container OOM killed when starting. I think these 2 tests is not to OOM kill, but to test `cgroup v2 resources.unified`, so let's increase the memory limit in thses 2 tests.

### Steps to reproduce the issue

1. open a PR against the main branch
2. wait CI running
3. CI failed

### Describe the results you received and expected

received:
```
not ok 18 runc run (cgroup v2 resources.unified only)
# (in test file tests/integration/cgroups.bats, line 218)
#   `[ "$status" -eq 0 ]' failed
# runc spec (status=0):
#
# runc run -d --console-socket /tmp/bats-run-jL79iM/runc.EGEZUn/tty/sock test_cgroups_unified (status=1):
# time="2023-09-16T18:26:34Z" level=warning msg="unable to get oom kill count" error="openat2 /sys/fs/cgroup/machine.slice/runc-cgroups-integration-test-18889.scope/memory.events: no such file or directory"
# time="2023-09-16T18:26:34Z" level=error msg="runc run failed: unable to start container process: error during container init: error setting cgroup config for procHooks process: failed to call BPF_PROG_ATTACH (BPF_CGROUP_DEVICE, BPF_F_ALLOW_MULTI): can't attach program: no such file or directory"
not ok 19 runc run (cgroup v2 resources.unified override)
# (in test file tests/integration/cgroups.bats, line 262)
#   `[ "$status" -eq 0 ]' failed
# runc spec (status=0):
#
# runc run -d --console-socket /tmp/bats-run-jL79iM/runc.I3CqFq/tty/sock test_cgroups_unified (status=1):
# time="2023-09-16T18:26:35Z" level=error msg="runc run failed: unable to start container process: container init was OOM-killed (memory limit too low?)"
```

expected:
CI should not fail.

### What version of runc are you using?

the main branch

### Host OS information

centos-stream-9

### Host kernel information

I can't see it.
